packages/byods/src/data-source-client/index.ts

```diff
@@ -171,13 +171,13 @@ export default class DataSourceClient {
    * @param {string} dataSourceId The id of the data source
    * @param {number} tokenLifetimeMinutes The refresh interval in minutes for the data source. Defaults to 60 mins. Should be <= 1440 & >=1.
    * @param {string} nonceGenerator Accepts an nonceGenerator that will generate nonce for the data source request.
-   * @returns {Promise<NodeJS.Timer>} A promise that resolves to the API response containing NodeJS.Timer.
+   * @returns {Promise<NodeJS.Timeout>} A promise that resolves to the API response containing NodeJS.Timer.
    */
   private async startAutoRefresh(
     dataSourceId: string,
     tokenLifetimeMinutes: number,
     nonceGenerator?: () => string
-  ): Promise<NodeJS.Timer> {
+  ): Promise<NodeJS.Timeout> {
```

Comment:
Nit: Had to update this in order to get the code to build.
As far as I tested it has not broken anything.
cc @pagour98

Reply:
Do we need to change and tests for this? @pagour98

Reply:
@Shreyas281299 
I think there is a difference between return type of setTimeout and setInterval here we have used setInterval so I returned the value as NodeJS.Timer, setTimeout return type is NodeJS.Timeout. If they both are compatible and things are working fine then we can change it to NodeJS.Timeout. There will be a slight change in one of the test case.